File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int timeSpentInOven) {
        return expectedMinutesInOven() - timeSpentInOven;
    }

    public int preparationTimeInMinutes(int layers) {
        return layers * 2;
    }

    public int totalTimeInMinutes(int layers, int timeSpentInOvens) {
        return preparationTimeInMinutes(layers) + timeSpentInOvens;
    }
}

File: solutions/elixir/lasagna/1/lib/lasagna.ex
defmodule Lasagna do
  # Please define the 'expected_minutes_in_oven/0' function
  @spec expected_minutes_in_oven() :: integer
  def expected_minutes_in_oven do
    40
  end

  # Please define the 'remaining_minutes_in_oven/1' function
  @spec remaining_minutes_in_oven(integer) :: integer
  def remaining_minutes_in_oven(actual_minutes_in_oven) do
    expected_minutes_in_oven() - actual_minutes_in_oven
  end

  # Please define the 'preparation_time_in_minutes/1' function
  @spec preparation_time_in_minutes(integer) :: integer
  def preparation_time_in_minutes(number_of_layers) do
    number_of_layers * 2
  end

  # Please define the 'total_time_in_minutes/2' function
  @spec total_time_in_minutes(integer, integer) :: integer
  def total_time_in_minutes(number_of_layers, actual_minutes_in_oven) do
    preparation_time_in_minutes(number_of_layers) + actual_minutes_in_oven
  end

  # Please define the 'alarm/0' function
  @spec alarm() :: String.t()
  def alarm do
    "Ding!"
  end
end
